### ISSUE TITLE
Not obvious how to remove nano

### ISSUE THREAD

**Issue by axelf4** [NixOS/nixpkgs#195795]
### Describe the bug
Pull request #53968 had the side effect of making it so that explicitly opting out of the nano package with
```
environment.defaultPackages = [];
```
no longer is sufficient for dropping the nano dependency from your NixOS system. `programs.nano.syntaxHighlighting` - which defaults to `true` - causes `/etc/nanorc` to be created which in turn references `pkgs.nano`.

Instead, I presume you would have to also say
```
programs.nano.syntaxHighlighting = false;
```
in addition to the above, which is not obvious as for why it should work.

Of course this is not that big of a deal, but nonetheless I can think of more fun ways of using that half a meg of storage space.

### Steps To Reproduce
1. On NixOS 22.05 do a rebuild ensuring that nano is not explicitly referenced, `environment.defaultPackages = []`, and `programs.nano.syntaxHighlighting` is unchanged.
2. Then there will be a store path for nano with (for some hash values)
   ```
   $ nix-store -q --referrers /nix/store/za7zbfk9lv6vwmyxzbar72z1b1xpw73a-nano-6.3
   /nix/store/za7zbfk9lv6vwmyxzbar72z1b1xpw73a-nano-6.3
   /nix/store/hdbn44axd02701zcapgaicvg80zv6ybj-etc-nanorc
   $ nix-store -q --referrers /nix/store/hdbn44axd02701zcapgaicvg80zv6ybj-etc-nanorc
   /nix/store/dlfl53xkzl5nk0w20hfc3pv652zsm334-etc
   ```
   and
   ```
   $ nix-store -q --referrers /nix/store/dlfl53xkzl5nk0w20hfc3pv652zsm334-etc
   /nix/store/vhmvkpy47nvyzcb9352ar2rvd0klgp52-nixos-system-axel-t450-22.05.20221011.0b20bf8
   ```
   which is the built NixOS system.

### Expected behavior
The NixOS system should not depend on nano when `environment.defaultPackages` does not include nano, unless otherwise requested.

### Screenshots
N/A

### Notify maintainers
@joachifm @Ma27

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.72, NixOS, 22.05 (Quokka), 22.05.20221011.0b20bf8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(axel): `""`
 - channels(root): `"nixos-21.11.334934.8a053bc2255, nixos-hardware"`
 - nixpkgs: `/nix/store/9djy37f9fgh6r3midb6h2467rp0hh0qi-source`
```

**Comment by yu-re-ka** [comment]
How about we move nano out of defaultPackages entirely, and instead have a `programs.nano.enable` that defaults to `true`?

**Comment by SFrijters** [comment]
I think this has recently been addressed via #244324.
